[Quan Teng Foong] iP

kaldius opened the pull request:
### Duke

> "Random quote here" - Quoted Person (Source here)


Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://www.google.com).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 💩

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

___
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

kaldius committed:
Add Greet, Echo, Exit

kaldius committed:
Add Task class

kaldius committed:
Tweak main function to record tasks

kaldius committed:
Add access modifier for Task.message

kaldius committed:
Add comments for Task class

kaldius committed:
Add "mark as done" functionality

kaldius committed:
Clean up code with helper functions

kaldius committed:
Add ToDo class

kaldius committed:
Add functionality to create ToDo object

kaldius committed:
Add Event class

kaldius committed:
Add functionality to create Event objects

kaldius committed:
Add Deadline class and deadline functionality

kaldius committed:
Cleanup (make tasks list a class variable, add an add function)

kaldius committed:
Add "Now you have x tasks in the list"

kaldius committed:
Add some tesets to EXPECTED.TXT and input.txt

kaldius committed:
Create NoSuchCommandException and move handling of commands to Task.java

kaldius committed:
Create TaskList class

kaldius committed:
Add handling of EmptyTaskListException

kaldius committed:
Add handling of NoTaskNameException

kaldius committed:
Add delete functionality

kaldius committed:
Fix for delete

kaldius committed:
Add storage class and necessary changes

kaldius committed:
Add saving and retrieving functionality

kaldius committed:
Add .gitignore for save.txt files

kaldius committed:
Add ability to use java.time.LocalDate for Deadline and Event

kaldius committed:
Merge branch 'branch-Level-7'

kaldius committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Task.java

kaldius committed:
Fix merge bugs

kaldius committed:
Fix more merge bugs

kaldius committed:
Add Ui class

kaldius committed:
Add Parser class

kaldius committed:
Improve Ui class

kaldius committed:
Add packages

kaldius committed:
Add more packages

kaldius committed:
Add javadocs

kaldius committed:
Add Command class

kaldius committed:
Modify duke to use commands

kaldius committed:
Add CreateNewToDoCommand

kaldius committed:
Add CreateNewEventCommand

kaldius committed:
Add CreateNewDeadlineCommand

kaldius committed:
Add TaskDoesNotExistException

kaldius committed:
Add JUnit test for Parser

kaldius committed:
Move ParserTest to duke package

kaldius committed:
Create JUnit test for Storage

kaldius committed:
Create JUnit tests for TaskList

kaldius committed:
Check and finalise JavaDocs

kaldius committed:
Add search method to TaskList

kaldius committed:
Add SearchCommand

kaldius committed:
Add JavaDocs

kaldius committed:
Add NoSearchResultsException

kaldius committed:
Merge branch 'branch-A-JavaDoc'

kaldius committed:
Merge branch 'branch-Level-9'

kaldius committed:
Fix bug to storage directory

qreoct commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,70 @@
+     * Makes a parent directory for the current filePath. If parent does not have a parent, this method
+     * recursively calls itself.
+     */
+    private void mkParentDir() {
```
Name for this method could use a verb instead, such as `makeParentDir()`

qreoct commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,112 @@
+            return new Event(message_and_eventDate[0], message_and_eventDate[1]);
+        case "deadline":
+            String[] message_and_endTime = input.split("/by ");
+            return new Deadline(message_and_endTime[0], message_and_endTime[1]);
```
According to the style guide _('Variable names must be in camelCase')_, it might be clearer to name these variables as `messageAndEventDate` instead.

qreoct commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# duke.Duke project template
```
Did you do an auto-refactor here using IntelliJ? It might have changed `Duke` to `duke.Duke` when you introduced packages, and it also affects non-code files such as in this README.

qreoct reviewed (commented):
I enjoyed reading your code! I've learned a thing or two for my own program, such as having a Command pattern to handle the various commands, as well as using `toStorage()` in your Task objects, which is really clever!

kaldius committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

kaldius committed:
Fix checkstyle mistakes

kaldius committed:
Merge branch 'branch-A-Gradle'

kaldius committed:
Create GUI according to guide

kaldius committed:
Tidy up GUI according to guide

kaldius committed:
Fix ShowListCommand to work with GUI, remaining Commands temporarily return ""

kaldius committed:
Fix CreateNewToDoCommand to work with GUI

kaldius committed:
Fix CreateNewEventCommand and CreateNewDeadlineCommand to work with GUI

kaldius committed:
Fix DeleteTaskCommand and SearchCommand to work with GUI

kaldius committed:
Fix SaveAndExitCommand and TaskCompletedCommand to work with GUI

kaldius committed:
Change user and duke display pics

kaldius committed:
Tidy up packages

NaokiShum commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,150 @@
+     *
+     * @return size of TaskList
+     */
+    public int size() {
```
Maybe the method name could be something like getSize() so that it will be more clear

NaokiShum commented on src/main/java/command/DeleteTaskCommand.java:
```diff
@@ -0,0 +1,23 @@
+import duke.TaskList;
+import duke.Ui;
+import exceptions.DukeException;
+
```
Missing javadocs for this file

NaokiShum reviewed (commented):
Code looks good overall with no real violation of the coding standard. You might have missed out on adding the javadocs for one of the file. Best of luck in the remaining part of the project!

kaldius committed:
Fix javadocs

kaldius committed:
Merge branch 'branch-Level-10'

kaldius committed:
Improve code quality

Repetition of code in the 3 CreateNew*Command classes

Abstract away reply string from the individual commands into
TaskList class such that it returns a success message each time
a new task is added.

This reduces code repetition and allows for easier tracing of
unintended outcomes in the TaskList::add success message.

kaldius committed:
TaskList class constructor: file argument assumed to exist

It is assumed that the storage class will return a file that exists.

Add assert statement that file exists.

If the file does not exist, there will be bad repercussions in the
rest of the program. It is thus important to check that it exists.

kaldius committed:
Merge pull request #2 from kaldius/branch-A-Assertions

Add assert statement in TaskList class

kaldius committed:
Merge pull request #1 from kaldius/branch-A-CodeQuality

Merge branch-A-code-quality

kaldius committed:
Merge branch 'branch-A-Assertions'

kaldius committed:
Add Recurring interface implemented by Event class

kaldius committed:
Add splitBy method to parser to split arguments and trim them

kaldius committed:
Add doTask for recurring events

kaldius committed:
Add ability to doTask and automatically create the next recurrence

kaldius committed:
Merge pull request #3 from kaldius/branch-B-RecurringTasks

Branch-B-RecurringTasks

kaldius committed:
Merge branch 'branch-B-RecurringTasks'

kaldius committed:
Exit README.md and add images

kaldius committed:
Fix README.md

kaldius committed:
Update duke.jar download link

kaldius committed:
Fix table formatting

kaldius committed:
Undo table formatting fix

kaldius committed:
Fix table formatting again
